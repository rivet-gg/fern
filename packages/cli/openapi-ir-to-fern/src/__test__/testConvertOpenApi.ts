import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { parse } from "@fern-api/openapi-parser";
import { createMockTaskContext } from "@fern-api/task-context";
import path from "path";
import { convert } from "../convert";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testConvertOpenAPI(fixtureName: string, filename: string, asyncApiFilename?: string): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("simple", async () => {
            const openApiPath = path.join(FIXTURES_PATH, fixtureName, filename);
            const mockTaskContext = createMockTaskContext({ logger: CONSOLE_LOGGER });

            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const specs = [];
            specs.push({
                absoluteFilepath: AbsoluteFilePath.of(openApiPath),
                absoluteFilepathToOverrides: undefined
            });
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const fernDefinition = convert({
                ir: openApiIr,
                taskContext: mockTaskContext,
                enableUniqueErrorsPerEndpoint: false,
                detectGlobalHeaders: true
            });
            expect(fernDefinition).toMatchSnapshot();
        });

        it("docs", async () => {
            const openApiPath = path.join(FIXTURES_PATH, fixtureName, filename);
            const mockTaskContext = createMockTaskContext({ logger: CONSOLE_LOGGER });

            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const specs = [];
            specs.push({
                absoluteFilepath: AbsoluteFilePath.of(openApiPath),
                absoluteFilepathToOverrides: undefined
            });
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const fernDefinition = convert({
                ir: openApiIr,
                taskContext: mockTaskContext,
                enableUniqueErrorsPerEndpoint: true,
                detectGlobalHeaders: false
            });
            expect(fernDefinition).toMatchSnapshot();
        });
    });
}
