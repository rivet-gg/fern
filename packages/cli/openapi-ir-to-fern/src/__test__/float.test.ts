import { testConvertOpenAPI } from "./testConvertOpenApi";

describe("float", () => {
    testConvertOpenAPI("float", "openapi.yml");
});
