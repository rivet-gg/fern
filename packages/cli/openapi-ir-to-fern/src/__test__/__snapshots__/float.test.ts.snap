// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`float float docs 1`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {
    "service": {
      "auth": false,
      "base-path": "",
      "endpoints": {
        "Get": {
          "auth": false,
          "docs": undefined,
          "examples": [
            {
              "request": {},
              "response": {
                "body": {
                  "id": 1.1,
                  "value": 1.1,
                },
              },
            },
          ],
          "method": "GET",
          "pagination": undefined,
          "path": "/get",
          "request": {
            "body": {
              "properties": {
                "id": "optional<float>",
              },
            },
            "headers": undefined,
            "name": "Request",
            "query-parameters": undefined,
          },
          "response": {
            "docs": "A simple API response.",
            "type": "Response",
          },
        },
      },
    },
    "types": {
      "Response": {
        "docs": "A generic response type used throughout the API.
",
        "properties": {
          "id": "optional<float>",
          "value": "optional<float>",
        },
      },
    },
  },
  "rootApiFile": {
    "display-name": "Test spec for \`float\` types specified as number formats.",
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;

exports[`float float simple 1`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {
    "service": {
      "auth": false,
      "base-path": "",
      "endpoints": {
        "Get": {
          "auth": false,
          "docs": undefined,
          "examples": [
            {
              "request": {},
              "response": {
                "body": {
                  "id": 1.1,
                  "value": 1.1,
                },
              },
            },
          ],
          "method": "GET",
          "pagination": undefined,
          "path": "/get",
          "request": {
            "body": {
              "properties": {
                "id": "optional<float>",
              },
            },
            "headers": undefined,
            "name": "Request",
            "query-parameters": undefined,
          },
          "response": {
            "docs": "A simple API response.",
            "type": "Response",
          },
        },
      },
    },
    "types": {
      "Response": {
        "docs": "A generic response type used throughout the API.
",
        "properties": {
          "id": "optional<float>",
          "value": "optional<float>",
        },
      },
    },
  },
  "rootApiFile": {
    "display-name": "Test spec for \`float\` types specified as number formats.",
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;
